jsonschema-core.xml

```diff
@@ -123,63 +123,109 @@
                 data in several ways.
             </t>
 
-            <section title="Assertions">
+            <section title="Keyword Behaviors">
                 <t>
-                    JSON Schema can be used to assert constraints on a JSON document, which
-                    either passes or fails the assertions.  This approach can be used to validate
-                    conformance with the constraints, or document what is needed to satisfy them.
+                    JSON Schema keywords fall into several general behavior categories.
+                    Assertions validate that an instance satisfies constraints, annotations
+                    attach information that applications may use in any way they see fit,
+                    and applicators allow for building more complex schemas than a single
+                    schema object would allow.
                 </t>
                 <t>
-                    JSON Schema implementations produce a single boolean result when evaluating
-                    an instance against schema assertions.
+                    Extension keywords SHOULD stay within these categories, keeping in mind
+                    that annotations in particular are extremely flexible.  Complex behavior
+                    is usually better delegated to applications on the basis of annotation
+                    data than implemented directly as schema keywords.  However, extension
+                    keywords MAY define other behaviors for specialized purposes.
                 </t>
-                <t>
-                    An instance can only fail an assertion that is present in the schema.
-                    In some cases, this no-op behavior is identical to a keyword that exists with
-                    certain values, and keyword defintions SHOULD note such values where known.
-                    These default behaviors MUST NOT result in assertion failures.
-                </t>
-                <section title="Assertions and Instance Primitive Types">
+                <section title="Applicators" anchor="applicators">
                     <t>
-                        Most assertions only constrain values within a certain
-                        primitive type.  When the type of the instance is not of the type
-                        targeted by the keyword, the instance is considered to conform
-                        to the assertion.
+                        Evaluation of an instance against a
```

Comment:
Is there a difference between "evaluation" and "application"?

Reply:
Hmm... yes that's unclear.  There is a difference, I'll add some text in the Keyword Behaviors intro section about it.